src/java.base/share/classes/sun/security/ssl/SSLKeyExchange.java

```diff
@@ -109,12 +124,14 @@ public SSLKeyDerivation createKeyDerivation(
     @Override
     public SSLHandshake[] getRelatedHandshakers(
             HandshakeContext handshakeContext) {
-        SSLHandshake[] auHandshakes;
+        SSLHandshake[] auHandshakes = null;
         if (authentication != null) {
-            auHandshakes =
-                authentication.getRelatedHandshakers(handshakeContext);
-        } else {
-            auHandshakes = null;
+            ListIterator<SSLAuthentication> li = authentication.listIterator();
+            while (li.hasNext() &&
+                    (auHandshakes == null || auHandshakes.length == 0)) {
+                SSLAuthentication authType = li.next();
+                auHandshakes = authType.getRelatedHandshakers(handshakeContext);
+            }
```

Comment:
I guess for-each loop could be a little bit more lightweight.

Reply:
Yeah, I thought about that.  I just like having all the loop termination clauses in one place rather than testing in the loop body and breaking.  I can change it to for-each though, no strong feelings one way or the other.